### PR TITLE
[stable/ark] Several improvements on naming, decoupling and metrics

### PR DESCRIPTION
**Pull request by PabloCastellano**
Ping @agolomoodysaada who was the last person who updated this chart

#### What this PR does / why we need it:

* Use new Ark naming:
    * Rename backupStorageProvider to backupStorageLocation
    * Rename persistentVolumeProvider to volumeSnapshotLocation
* Add configuration.provider so that backupStorageLocation is decoupled from
  the cloud provider where ark it deployed to (hence we can have multiple backup locations in different cloud providers)
* Fixed typo in metricsAddress (now renamed to configuration.metrics.address)
* Expose metrics port with configuration.metrics.enabled
* Support new prefix parameter for backupStorageLocation

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Review by agolomoodysaada — APPROVED** [review]
LGTM! Thank you  🎉

**Comment by agolomoodysaada** [comment]
@cpanato

File: stable/ark/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 0.10.1
 description: A Helm chart for ark
 name: ark
-version: 3.0.0
+version: 3.0.1
```
Comment (unguiculus) [inline stable/ark/Chart.yaml]: Seems like this is a breaking change which needs a major version bump.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/ark/Chart.yaml (hunk 1 above)
Comment (PabloCastellano) [reply stable/ark/Chart.yaml]: Agree, changed!

**Comment by PabloCastellano** [comment]
I'm wondering if we should also merge changes from https://github.com/helm/charts/pull/9496 together to avoid another breaking change and major version bump. @JoschaLaubach would you be OK if I do it?

**Comment by JoschaLaubach** [comment]
@PabloCastellano Sure thing, go ahead.

**Comment by cpanato** [comment]
@PabloCastellano please update your branch thanks!

**Comment by PabloCastellano** [comment]
Updated now. The e2e test ran and failed due to the following error

```
Error: customresourcedefinitions.apiextensions.k8s.io "downloadrequests.ark.heptio.com" already exists
```

Is it related to https://github.com/helm/helm/issues/4697 ? 
According to https://github.com/helm/charts/pull/7165#issuecomment-437375838 this is a known issue for stable/ark already

**Comment by cpanato** [comment]
@PabloCastellano looks like this is the issue, i will cleanup manually

**Review by unguiculus — APPROVED** [review]

**Pull request merged**